### PR TITLE
Some performance tweaks

### PR DESCRIPTION
**Pull request by annando**
This includes index improvements, caching of the function `getOwnerDataById` and replacing the function `Profile::getByNickname` with `getOwnerDataById`.

### CODE REVIEW

File: src/Model/User.php [hunk 1]
```diff
@@ -213,6 +215,10 @@ public static function getFirstAdmin(array $fields = [])
 	 */
 	public static function getOwnerDataById($uid, $check_valid = true)
 	{
+		if (!empty(self::$owner)) {
+			return self::$owner;
```
Comment (MrPetovan) [inline src/Model/User.php:219]: I can't accept this, it looks like a session cache poisoning footgun. Please use a cache keyed by `$uid`.

Comment (annando) [reply src/Model/User.php:219]: Örgs, you are right.

Comment (annando) [reply src/Model/User.php:219]: This possibly explains some weird behaviour I observed.

Comment (annando) [reply src/Model/User.php]: It is changed into an array.

**Pull request merged**